src/modules/ekf2/EKF/control.cpp

```diff
@@ -848,7 +848,7 @@ void Ekf::controlHeightSensorTimeouts()
 
 	// check if height has been inertial deadreckoning for too long
 	// in vision hgt mode check for vision data
-	const bool hgt_fusion_timeout = isTimedOut(_time_last_hgt_fuse, (uint64_t)5e6);
+	const bool hgt_fusion_timeout = isTimedOut(_time_last_hgt_fuse, (uint64_t)3e5);
```

Comment:
why?

Reply:
A fallback does only make sense, if operation continues with as little impact as possible. Therefore the timeout should be detected as soon as possible.

Reply:
Well, yes, but no height fusion doesn't mean no vertical velocity aiding. In case of GNSS vel + baro height fusion, you don't want to trigger a height reset on a wind gust for example. The rest of the PR is quite specific to EV aiding but this timeout is much more general and we should be careful when changing them. 
I would suggest to remove that change such that we can merge the PR and then you can make another one to change the timeout. An option would be to set a parameter for it if in you really need to have a smaller timeout threshold for your application; we can then easily merge it without risking to change the behavior of other existing setups where the current value is maybe better.

Reply:
My proposal would be to make the timeout dependent on the selected height source, keeping it at 5secs for other sources than EV or RF.